renderer/modules/shared/components/drawer/navigation/custom-drawer/custom-drawer.tsx

```diff
@@ -1,17 +1,103 @@
+import { useState, useEffect } from 'react';
 import { useRouter } from 'next/router';
 import { drawerLinks, DrawerLink } from './drawer-links';
 import List from '@material-ui/core/List';
 import SectionLink from '../section-link/section-link';
 import useDawerStyles from '../../drawer.style';
-import UpgradeBanner from '../../../upgrade-banner/upgrade-banner';
+import UpgradeBanner, {
+  UpgradeBannerProps,
+} from '../../../upgrade-banner/upgrade-banner';
 import ProfilePicture from '../../../profile-picture/profile-picture';
+import { IpcRendererEvent } from 'electron';
 
 interface CustomDrawerProps {
   classes: ReturnType<typeof useDawerStyles>;
 }
 
+interface IdeDistribution {
+  id: string;
+  ideConfig: {
+    basepath: string;
+    workspaces: string;
+    commands: string;
+    version: string;
+  };
+}
+
+interface DevonIdeScripts {
+  version: string;
+  updated: Date;
+}
 export default function CustomDrawer(props: CustomDrawerProps): JSX.Element {
   const router = useRouter();
+  const initialProps: UpgradeBannerProps = {
+    version: '',
+    infoText: '',
+  };
+  const [displayUpgradeBanner, setDisplayUpgradeBanner] = useState(false);
+  const [upgradeBannerProps, setUpgradeBannerProps] = useState<
+    UpgradeBannerProps
+  >(initialProps);
+
+  const getLocalLatestVersion = (versions: string[]): string => {
+    const numericVersions = versions.map((v) => {
+      return +v.split('.').join(''); // for input "2020.04.001" it will return 202004001
+    });
```

Comment:
You dont need to do this. You can directly compare string

"2020.1" > "2020.2" -> false
"2020.2" > "2020.1" -> true

Math.max works with strings too.

Reply:
Math.max will work for strings which look like numbers, like "2020.1". It won't work for our use case where we have version numbers like "2020.04.001"

Reply:
Yes, you are right with the math.max thing
Other option is using reducer

['2020.04.001', '2020.04.002', '2020.04.003'].reduce(
    (currentVersion, maxVersion) =>
      currentVersion > maxVersion? currentVersion: maxVersion,
    ''
  );

// Output 2020.04.003